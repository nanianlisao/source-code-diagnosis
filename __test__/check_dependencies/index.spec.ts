import { expect, test, } from 'vitest'
import { fileURLToPath } from "node:url";
import path, { dirname } from "node:path";
import { checkDependencies, DependencyNode} from '../../index.js'



const __filename = fileURLToPath(import.meta.url);

function normalizePaths(node: DependencyNode): DependencyNode {
  const basePath = dirname(__filename);
  return {
    name: path.relative(basePath, node.name).replace(/\\/g, '/'),
    children: node.children.map(normalizePaths),
    astNode: node.astNode
  };
}


test('The dependency file list of the specified file should be obtained normally.', () => {
  const cwd = path.resolve(dirname(__filename),"features","normal");
  const response = checkDependencies(path.join(cwd,"c.js"),{
    cwd,
  })

  const normalizedPaths = normalizePaths(response);

  expect(normalizedPaths).toMatchSnapshot()
 
})

test('The dependency file list of the specified file should be obtained normally.', () => {
  const cwd = path.resolve(dirname(__filename),"features","alias");
  const response = checkDependencies(path.join(cwd,"c.js"),{
    alias:{
      "@":[cwd]
    },
    cwd,
  })

  const normalizedPaths = normalizePaths(response);

  expect(normalizedPaths).toMatchSnapshot()
 


})