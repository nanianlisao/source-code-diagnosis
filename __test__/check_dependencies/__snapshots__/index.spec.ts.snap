// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The dependency file list of the specified file should be obtained normally. 1`] = `
{
  "astNode": {
    "loc": {
      "end": {
        "col": 25,
        "line": 1,
      },
      "start": {
        "col": 1,
        "line": 1,
      },
    },
    "span": {
      "end": 24,
      "start": 0,
    },
  },
  "children": [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "children": [
        {
          "astNode": undefined,
          "children": [],
          "name": "features/normal/utils.js",
        },
      ],
      "name": "features/normal/a.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "children": [],
      "name": "features/normal/b.js",
    },
  ],
  "name": "features/normal/c.js",
}
`;

exports[`The dependency file list of the specified file should be obtained normally. 2`] = `
{
  "astNode": {
    "loc": {
      "end": {
        "col": 25,
        "line": 1,
      },
      "start": {
        "col": 1,
        "line": 1,
      },
    },
    "span": {
      "end": 24,
      "start": 0,
    },
  },
  "children": [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "children": [
        {
          "astNode": undefined,
          "children": [],
          "name": "features/alias/utils.js",
        },
      ],
      "name": "features/alias/a.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "children": [],
      "name": "features/alias/b.js",
    },
  ],
  "name": "features/alias/c.js",
}
`;
